lib/presentation/view/order_by/order_by.dart

```diff
@@ -0,0 +1,93 @@
+// provided you comply with the Additional Terms applicable for LinShare software by
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+import 'package:domain/domain.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:linshare_flutter_app/presentation/di/get_it_service.dart';
+import 'package:linshare_flutter_app/presentation/util/app_image_paths.dart';
+import 'package:linshare_flutter_app/presentation/util/extensions/color_extension.dart';
+import 'package:linshare_flutter_app/presentation/util/extensions/order_by_extension.dart';
+import 'package:linshare_flutter_app/presentation/util/extensions/order_type_extension.dart';
+
+typedef OnOpenOrderMenuActionClick = void Function(Sorter sorter);
+
+class OrderByBuilder {
+  final imagePath = getIt<AppImagePaths>();
+
+  final BuildContext _context;
+
+  Sorter _sorter;
+  OnOpenOrderMenuActionClick _onOpenOrderMenuActionClick;
+
+  OrderByBuilder(this._context);
+
+  OrderByBuilder sorter(Sorter sorter) {
+    _sorter = sorter;
+    return this;
+  }
+
+  OrderByBuilder onOpenOrderMenuAction(
+      OnOpenOrderMenuActionClick onOpenOrderMenuActionClick) {
+    _onOpenOrderMenuActionClick = onOpenOrderMenuActionClick;
+    return this;
+  }
+
+  Widget build() {
+    return _buildMenuSorter();
+  }
+
+  Widget _buildMenuSorter() {
```

Comment:
This should return ListTile so it's more specific

Reply:
Cool, it fixed, it